lib/Sema/TypeCheckStorage.cpp

```diff
@@ -2324,6 +2324,17 @@ createCoroutineAccessorPrototype(AbstractStorageDecl *storage,
   AvailabilityInference::applyInferredAvailableAttrs(accessor,
                                                      asAvailableAs, ctx);
 
+  // A modify coroutine should have the same SPI visibility as the setter.
```

Comment:
Can we fold this into applyInferredAvailableAttrs()?

Reply:
I think it's worth thinking about but I don't think that would be a low risk change and I'd like to have more than one concrete known problem that would be solved by making that change before I pursued generalizing like that. It's not clear to me that `@_spi` and `@available` ought to follow the same logic. For example, we'd need a merging algorithm to handle when the inherited-from decls have distinct SPI groups. I'm not sure whether the more correct general approach to that would be to apply the union, the intersection, or neither.

Reply:
After thinking about it more, I've concluded that it's not always the case that the set of decls you want to infer SPI status from are always the same set of decls you want to infer availability from. However, I can at least pull the logic for applying SPI attrs out into its own function so that it's easy to fix a similar problem for other synthesized decls in the future.

Reply:
No worries — it was just a random suggestion.